dvc/commands/experiments/push.py

```diff
@@ -18,34 +19,60 @@ def raise_error_if_all_disabled(self):
                 "`--rev` or `--all-commits` flag."
             )
 
-    def run(self):
+    @staticmethod
+    def log_result(result: Dict[str, Any], remote: str):
         from dvc.utils import humanize
 
-        self.raise_error_if_all_disabled()
+        def join_exps(exps):
+            return humanize.join([f"[bold]{e}[/]" for e in exps])
 
-        result = self.repo.experiments.push(
-            self.args.git_remote,
-            self.args.experiment,
-            all_commits=self.args.all_commits,
-            rev=self.args.rev,
-            num=self.args.num,
-            force=self.args.force,
-            push_cache=self.args.push_cache,
-            dvc_remote=self.args.dvc_remote,
-            jobs=self.args.jobs,
-            run_cache=self.args.run_cache,
-        )
-        if pushed_exps := result.get("pushed"):
-            ui.write(
-                f"Pushed experiment {humanize.join(map(repr, pushed_exps))} "
-                f"to Git remote {self.args.git_remote!r}."
+        if diverged_exps := result.get("diverged"):
+            exps = join_exps(diverged_exps)
+            ui.warn(
+                f"Local experiment {exps} has diverged "
+                "from remote experiment with the same name. "
+                "To override the remote experiment re-run with '--force'."
             )
-        else:
+        if uptodate_exps := result.get("up_to_date"):
+            verb = "are" if len(uptodate_exps) > 1 else "is"
+            exps = join_exps(uptodate_exps)
+            ui.write(f"Experiment {exps} {verb} up to date.", styled=True)
+        if pushed_exps := result.get("success"):
+            exps = join_exps(pushed_exps)
+            ui.write(f"Pushed experiment {exps} to Git remote {remote!r}.", styled=True)
+        if not uptodate_exps and not pushed_exps:
             ui.write("No experiments to push.")
 
         if project_url := result.get("url"):
             ui.write("[yellow]View your experiments at", project_url, styled=True)
```

Comment:
Not a blocker if it's hard to get the URL in this scenario, but could we show it even when nothing new gets pushed? If I was trying to push an exp that was pushed already, I still may want to see it in Studio or see which experiments are there already.

Reply:
We did this so that we don’t spam Studio too much. Maybe we should pretend as if we pushed something here, lie to Studio.